StepperEngine/src/Flow/Flow.java

```diff
@@ -242,4 +241,30 @@ private void addOutputs(List<DataType> outputsToAdd) {
     public String getName() {
         return name;
     }
+
+    public void execution(ArrayList<DataType> inputs){
+        //might need to allocate to the inputs within the steps
+        for(DataType input:inputs){
+            for(DataType freeInput:freeInputs.get(input.getEffectiveName()))
+                freeInput.setData(input);
+        }
+
+        try {
+            for (Step step : steps) {
+                step.execute();
+                if (step.getStatus() == Step.Status.Failure)
```

Comment:
But some steps can fail (have Failure status) and not fail the whole Flow, no?

Reply:
yea i alr eady worked on it on my next pr